### ISSUE TITLE
As a user I would like to have direct message history

### ISSUE THREAD

**Issue by poixen** [Cockatrice/Cockatrice#2146]
The history of chats is stored on the server, we can use this the same way we use for the main chat. Have previous chat in a light grey.

**Comment by tooomm** [comment]
Like you enter the server and start a private chat with a mate. The server would then push the last x messages from the former conversation to you?

(related https://github.com/Cockatrice/Cockatrice/issues/1532)

**Comment by Jinxstands4Jinx** [comment]
How would this work overtime? like if I started a conversation with a player and a couple of days later I retalk to that player would the old conversation lines carry over?

**Comment by ZeldaZach** [comment]
I'm a big fan of this, and I somewhat like how Steam/Valve does it. They show you a few messages (~10) from the past ~5 days in case you haven't spoken since then and you can review your past chats like that. I'd be fine with something like this for sure!

**Comment by ThomasDautzenberg** [comment]
I think just randomly saving the last 5-10 days might be frustrating to some, who don't know there is a cap and start using the chat to store information. (e.g. for decktech discussion)

I'd suggest a checkbox 'save this conversation' which, when checked, stores the conversation history on your computer.

**Comment by skwerlman** [comment]
Rather than saving per-conversation, why not have a toggle in settings between Steam-style backlogs and full conversation logging? This is feature's optimal behavior is very much a matter of preference.

**Comment by Daenyth** [comment]
I don't think we should put that burden on server ops; chat logs and up taking up a fair amount of storage and we should be able to set reasonable upper bounds for moderators. I think keeping the past week of chat would be a fine amount.

**Comment by Daenyth** [comment]
Also in general I dislike adding new toggles because users already get confused by the amount of config and features we have. It's easier to explain if it always behaves the same.

**Comment by tooomm** [comment]
In https://github.com/Cockatrice/Cockatrice/issues/1532 the idea of a local storage of former messages got mentioned, too.

**Comment by ZeldaZach** [comment]
Local storage would be an interesting way to go about this. No way I would do this server sided

**Comment by ZeldaZach** [comment]
Looking back, a server sided approach might be fine. It can pull from whatever chat is there, it's no burden if the server op chooses not to support chat logs.
